src/main/java/umaikaze/duke/Duke.java

```diff
@@ -0,0 +1,74 @@
+package umaikaze.duke;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.UnsupportedEncodingException;
+import java.time.format.DateTimeParseException;
+
+public class Duke {
+    Ui ui;
+    Storage st;
+    TaskList tl;
+
+    public Duke(String savePath) throws UnsupportedEncodingException {
+        ui = new Ui();
+        st = new Storage(savePath);
+        try {
+            tl = new TaskList(st.loadFile());
+        } catch (IOException e) {
+            ui.showError(e.getMessage());
+            tl = new TaskList();
+        }
+    }
+
+    void run(BufferedReader br) throws IOException {
```

Comment:
Very long method, consider abstracting some parts of it into another separate method, e.g. the `switch` block and the part about saving to file.